clippy_lints/src/doc/too_long_first_doc_paragraph.rs

```diff
@@ -0,0 +1,59 @@
+use rustc_ast::ast::Attribute;
+use rustc_hir::ItemKind;
+use rustc_lint::LateContext;
+
+use clippy_utils::diagnostics::span_lint;
+
+use super::TOO_LONG_FIRST_DOC_PARAGRAPH;
+
+pub(super) fn check(
+    cx: &LateContext<'_>,
+    attrs: &[Attribute],
+    item_kind: ItemKind<'_>,
+    mut first_paragraph_len: usize,
+) {
+    if first_paragraph_len <= 80
```

Comment:
Just being curious 🤔~ why limit it to 80? Is it some kind of standard? Should it be mentioned on the lint documentation or having a configuration otherwise?

Reply:
Not sure if having a config would improve things much. This lint is mostly to prevent things like:

![image](https://github.com/rust-lang/rust-clippy/assets/3050060/0fd68569-a575-4848-b53f-376a1a7d6049)

Reply:
80 seems *way* too low for a default limit, if there's only one line: I don't think we should emit a warning [here](https://github.com/rust-lang/rust-clippy/blob/863179081ee12284349b6356e7bb444bcfde2846/tests/ui/mixed_attributes_style/mod_declaration.rs#L2).
This looks fine to me, there's nothing to split up here really:

![image](https://github.com/rust-lang/rust-clippy/assets/30553356/2cd1ec8f-9c25-4f75-bf6d-e838cad438a3)

Reply:
We can take a bigger limit. But in any case, the first line is supposed to be a summary, not a description. But I don't consider this a universal thing, hence why I put it as allow by default.

Reply:
I think a good limit would be whenever it splits it on a 1080p/720p monitor in fullscreen mode on the browser (that way it's consistent)

Reply:
> hence why I put it as allow by default.

It's currently `style`, but IMO it should stay like that

Reply:
It's worth referring back to what the rustdoc book says about this:

> It is recommended that each item's documentation follows this basic structure:

> [short sentence explaining what it is]

> [more detailed explanation]

> [at least one code example that users can copy/paste to try it]

> [even more advanced explanations if necessary]

> Everything before the first empty line will be reused to describe the component in searches and module overviews. For example, the function std::env::args() above will be shown on the [std::env](https://doc.rust-lang.org/stable/std/env/index.html#functions) module documentation. It is good practice to keep the summary to one line: concise writing is a goal of good documentation.
—https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html

The `std` documentation sticks to this pretty well; it's mostly one short sentence (sometimes two: "Removes and returns the last element in the map. The key of this element is the maximum key that was in the map.") On the other hand, sometimes the 'more detailed explanation' paragraph is joined on to the short intro sentence when it doesn't need to be.

So, how long is "one line"? For what it's worth, on my system `doc.rust-lang.org` wraps the text at 132 characters per line, so what about warning at something like 250 characters? That would in any case be a fairly long sentence. 

@GuillaumeGomez, you could try running your lint over the `std` documentation with this setting and see how often it trips?

Reply:
If your type is named `SomethinVeryLongAndBlaBlaBlaBlaBlaBlaBlaBlaBlaBlaBlaBlaBlaBlaBlaBlaBlaBlaBlaBla`, it will wrap the text at much less, hence why I picked 80 as a default. (It's an issue in rustdoc that needs to be fixed too.)

Big +1 for copying the explanations from the book.

> @GuillaumeGomez, you could try running your lint over the `std` documentation with this setting and see how often it trips?

I have no idea how to run a local clippy on anything else. If you know how or the location explaining how, it'd be very useful. ;)

Reply:
> I have no idea how to run a local clippy on anything else. If you know how or the location explaining how, it'd be very useful. ;)


I don't know about running it on the standard library specifically, since it's special, but for normal cargo projects you can just run `cargo dev lint path/to/cargo/project` and it should just work.


`std::env` in particular has 13 out of 23 items that have >80 characters in their first paragraph. `slice` has 169 characters.
This rate seems fairly high for the standard library if we were to emit a warning for all of those.

I also did a lintcheck run on the top 200 popular crates and there are 2008 warnings in total of this lint.

Reply:
> > I have no idea how to run a local clippy on anything else. If you know how or the location explaining how, it'd be very useful. ;)
> 
> I don't know about running it on the standard library specifically, since it's special, but for normal cargo projects you can just run `cargo dev lint path/to/cargo/project` and it should just work.

Nice, thanks!

> `std::env` in particular has 13 out of 23 items that have >80 characters in their first paragraph. `slice` has 169 characters. This rate seems fairly high for the standard library if we were to emit a warning for all of those.
> 
> I also did a lintcheck run on the top 200 popular crates and there are 2008 warnings in total of this lint.

Well, unless I miss something, `style` is `allow` by default, no? If so should be fine. Would still be better to not emit this many warnings though...